### PR TITLE
WIP: Fix CircleCI

### PR DESCRIPTION
**Pull request by larsoner**
CircleCI is failing on a couple examples on `main` and a recent PR:

https://app.circleci.com/pipelines/github/mne-tools/mne-python/13057/workflows/73fd5c5d-72f1-4efb-be9a-7ba7892b8b7c/jobs/42068
https://app.circleci.com/pipelines/github/mne-tools/mne-python/13070/workflows/a2c08a05-2b91-4c82-ad77-4356b1ca0560/jobs/42097

I hope this fixes the error. We'll see!

### CODE REVIEW

File: mne/viz/backends/_qt.py [hunk 1]
```diff
@@ -545,6 +545,7 @@ def _window_initialize(self):
         self._window.signal_close.connect(self._window_clean)
         self._window_close_callbacks = list()
 
+        """
```
Comment (larsoner) [inline mne/viz/backends/_qt.py:548]: @GuillaumeFavelier locally with this block "commented out", this passes:
```
PATTERN=10_ieeg make html_dev-pattern
```
but with it in place, I get a failure:


So this makes me think that the CircleCI failure on `main` is due to https://github.com/mne-tools/mne-python/pull/10305

Can you take a look and see if:

1. you can replicate locally that on `main` the command above fails the same way
2. you can replicate that things work on this branch

If that's the case, can you look into a solution here that works with the dialog stuff?

CircleCI at least seems to agree with this assessment of the problem, as the example passes with this commented out:

https://app.circleci.com/pipelines/github/mne-tools/mne-python/13075/workflows/53035c2b-efc6-4833-93ae-c74c9b015f57/jobs/42110

Comment (larsoner) [reply mne/viz/backends/_qt.py:548]: ... FYI @GuillaumeFavelier I spent a lot of time trying to resolve the issues but couldn't do it :( I must be missing something.

Comment (GuillaumeFavelier) [reply mne/viz/backends/_qt.py:548]: I can reproduce locally @larsoner. Indeed it seems related to the new `close_connect` code but the error is in Brain though:

<details>

```
Traceback (most recent call last):
  File "/home/guillaume/source/anaconda3/envs/mne-py38/lib/python3.8/site-packages/sphinx/events.py", line 101, in emit
    results.append(listener.handler(self.app, *args))
  File "/home/guillaume/source/sphinx-gallery/sphinx_gallery/gen_gallery.py", line 464, in generate_gallery_rst
    generate_dir_rst(src_dir, target_dir, gallery_conf,
  File "/home/guillaume/source/sphinx-gallery/sphinx_gallery/gen_rst.py", line 399, in generate_dir_rst
    intro, title, cost = generate_file_rst(
  File "/home/guillaume/source/sphinx-gallery/sphinx_gallery/gen_rst.py", line 1031, in generate_file_rst
    clean_modules(gallery_conf, fname, 'after')
  File "/home/guillaume/source/sphinx-gallery/sphinx_gallery/scrapers.py", line 596, in clean_modules
    reset_module(gallery_conf, fname, when=when)
  File "/home/guillaume/source/mne-python/doc/conf.py", line 341, in __call__
    _assert_no_instances(Brain, when)  # calls gc.collect()
  File "/home/guillaume/source/mne-python/mne/utils/misc.py", line 368, in _assert_no_instances
    assert n == 0, f'\n{n} {cls.__name__} @ {when}:\n' + '\n'.join(ref)
AssertionError: 
4 Brain @ mne/conf.py:Resetter.__call__:after:10_ieeg_localize.py:
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f6987b9c460>>
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f698d666b80>>
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f6987b0f520>>
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f698d6665e0>>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/guillaume/source/anaconda3/envs/mne-py38/lib/python3.8/site-packages/sphinx/cmd/build.py", line 276, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/home/guillaume/source/anaconda3/envs/mne-py38/lib/python3.8/site-packages/sphinx/application.py", line 270, in __init__
    self._init_builder()
  File "/home/guillaume/source/anaconda3/envs/mne-py38/lib/python3.8/site-packages/sphinx/application.py", line 328, in _init_builder
    self.events.emit('builder-inited')
  File "/home/guillaume/source/anaconda3/envs/mne-py38/lib/python3.8/site-packages/sphinx/events.py", line 109, in emit
    raise ExtensionError(__("Handler %r for event %r threw an exception") %
sphinx.errors.ExtensionError: Handler <function generate_gallery_rst at 0x7f69f256de50> for event 'builder-inited' threw an exception (exception: 
4 Brain @ mne/conf.py:Resetter.__call__:after:10_ieeg_localize.py:
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f6987b9c460>>
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f698d666b80>>
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f6987b0f520>>
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f698d6665e0>>)

Extension error (sphinx_gallery.gen_gallery):
Handler <function generate_gallery_rst at 0x7f69f256de50> for event 'builder-inited' threw an exception (exception: 
4 Brain @ mne/conf.py:Resetter.__call__:after:10_ieeg_localize.py:
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f6987b9c460>>
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f698d666b80>>
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f6987b0f520>>
Brain._cleaned = True
method: <bound method Brain._clean of <mne.viz._brain._brain.Brain object at 0x7f698d6665e0>>)
make: *** [Makefile:60: html_dev-pattern] Error 2
```

</details>

Comment (GuillaumeFavelier) [reply mne/viz/backends/_qt.py:548]: I opened https://github.com/mne-tools/mne-python/pull/10388

**Comment by larsoner** [comment]
Closing for #10388

**Pull request closed without merging**